Use the ctx provided when invoking a plugin

mach6 opened the pull request:
Plugin implementations pass a ctx when
invoking each plugin. This was previously
ignored.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/drone/drone-go?pullRequest=52) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/drone/drone-go?pullRequest=52) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/drone/drone-go?pullRequest=52) it.</sub>

mach6 commented:
> Now all calls are losing the pre-configured timeout, but otherwise LGTM

My assumption is the caller is passing a proper context established -- which is the case with the `drone` server calls. I know runners also call into `drone-go`, e.g. `environ` and `secret` plugins . I'm assuming they also are setting a timeout, when needed.

Is this a bad assumption?

bradrydzewski commented:
> My assumption is the caller is passing a proper context 

this is something we should double-check but the runner-go library [sets a context timeout](https://github.com/drone/runner-go/blob/master/registry/external.go#L42) for registry extension calls. We should make sure this is the case for all extension calls. Assuming that is the case, I think we could just change the `Do` signature since the client is internal use only.

```diff
+func (s *Client) Do(ctx context.Context, in, out interface{}) error {
-func (s *Client) Do(in, out interface{}) error {
-	ctx := context.Background()
-	ctx, cancel := context.WithTimeout(ctx, time.Minute)
-	defer cancel()
-
	return s.DoUsingContext(ctx, in, out)
}
```

tboerger commented:
Looking at https://github.com/drone/drone/pull/2998 I think drone is already defining timeouts.

mach6 commented:
> I think we could just change the `Do` signature since the client is internal use only.

Sure. Happy to change the signature, instead. LMK.

mach6 commented:
`runner-go` also sets timeout for `environ` and `secret`

mach6 commented:
@tboerger @bradrydzewski -- I went ahead and changed the signature of `(s *Client) Do` and dropped the use of the fixed `context.WithTimeout(time.Minute)` in the latest commit. I think we are covered on the question of callers setting a context timeout.
